core/dbt/graph/queue.py

```diff
@@ -212,3 +216,80 @@ def wait_until_something_was_done(self) -> int:
         with self.lock:
             self.some_task_done.wait()
             return self.inner.unfinished_tasks
+
+
+class ExecutionQueue:
```

Comment:
@MichelleArk @gshank This is the interface I am thinking of moving our queue to, the selection part will be hidden here so from the task perspective it just constructs the queue and executes nodes got from the queue.